colossalai/zero/sharded_param/sharded_param.py

```diff
@@ -5,18 +5,13 @@
 from .tensorful_state import StatefulTensor, TensorState
 from typing import List
 
-# use this tensor as empty data point for parameters
-# we do not want users use param.data when its torch payload is removed
-# empty tensor is expected to raise error when get used
-FAKE_EMPTY_TENSOR = torch.BoolTensor([], device='cpu')
-
 EMPTY_TENSOR_DICT = {}
 
 
 def get_empty_tensor(device: torch.device, dtype: torch.dtype):
     key = (device, dtype)
     if key not in EMPTY_TENSOR_DICT:
-        EMPTY_TENSOR_DICT[key] = FAKE_EMPTY_TENSOR.to(device, dtype)
+        EMPTY_TENSOR_DICT[key] = torch.empty(0, dtype=dtype, device=device)
```

Comment:
The original code. different keys share the same memory space if the device and type are the same.
Now, for each key has an individual memory space.
@ver217

Reply:
This is used to acoid wasted memory in cpu.

Reply:
OK